### ISSUE TITLE
TextBlockGoogleStyleFormatting: opening """ preceded by ? is not flagged as violation

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#19707]
I have read check documentation: https://checkstyle.org/checks/coding/textblockgooglestyleformatting.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/TernaryBug.java << 'EOF'
class TernaryBug {
    boolean flag = true;

    String a = flag
        ? """
          yes
          """
        : """
          no
          """;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/ternary_check.xml << 'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="TextBlockGoogleStyleFormatting"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % java -jar target/checkstyle-13.4.1-SNAPSHOT-all.jar -c /tmp/ternary_check.xml /tmp/TernaryBug.java

Starting audit...
[ERROR] /tmp/TernaryBug.java:8:11: Opening quotes (""") of text-block must be on the new line. [TextBlockGoogleStyleFormatting]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Expected behavior:**

Both `? """`  and `: """`  have an operator preceding `"""` on the same line, so both should be flagged. 

Currently only the `:` branch  is flagged. The `?` branch  is missed because `openingQuotesAreAloneOnTheLine()` compares `"""` against `openingQuotes.getPreviousSibling()`, which for the `?` branch is the condition expression (`flag` on different line), not the `?` operator. The `?` token is the parent `QUESTION` node, not a sibling, so the same-line check never sees it. When the condition is on the same line as `? """` (e.g., `flag ? """`), the check works by accident.

**Comment by vivek-0509** [comment]
i am on it

**Comment by vivek-0509** [comment]
@romani Please take a look

**Comment by romani** [comment]
@vivek-0509 , please share google formatter output and diff on this Input file.
To be sure we are not conflicting with auto formater.

**Comment by vivek-0509** [comment]
@romani here is the google-java-format output and diff 
```
vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/TernaryBug.java << 'EOF'
class TernaryBug {
    boolean flag = true;

    String a = flag
        ? """
          yes
          """
        : """
          no
          """;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java -jar /tmp/google-java-format.jar /tmp/TernaryBug.java > /tmp/TernaryBug_formatted.java
diff -u /tmp/TernaryBug.java /tmp/TernaryBug_formatted.java
--- /tmp/TernaryBug.java        2026-04-18 18:43:21
+++ /tmp/TernaryBug_formatted.java      2026-04-18 18:43:26
@@ -1,11 +1,12 @@

 class TernaryBug {
-    boolean flag = true;
+  boolean flag = true;
 
-    String a = flag
-        ? """
+  String a =
+      flag
+          ? """
           yes
           """
-        : """
+          : """
           no
           """;
 }

vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/TernaryBug.java << 'EOF'
class TernaryBug {
    boolean flag = true;

    String d = flag ?
        """
        yes
        """
        :
        """
        no
        """;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java -jar /tmp/google-java-format.jar /tmp/TernaryBug.java > /tmp/TernaryBug_formatted.java
diff -u /tmp/TernaryBug.java /tmp/TernaryBug_formatted.java
--- /tmp/TernaryBug.java        2026-04-18 18:45:14
+++ /tmp/TernaryBug_formatted.java      2026-04-18 18:45:19
@@ -1,12 +1,12 @@

 class TernaryBug {
-    boolean flag = true;
+  boolean flag = true;
 
-    String d = flag ?
-        """
-        yes
-        """
-        :
-        """
-        no
-        """;
+  String d =
+      flag
+          ? """
+          yes
+          """
+          : """
+          no
+          """;
 }
```

The Google Java Style Guide [Section 4.8.9  Text Blocks](https://google.github.io/styleguide/javaguide.html#s4.8.9-text-blocks) states:

> *The opening `"""` of a text block is always on a new line.*

According to this rule, both `? """` and `: """` have the opening `"""` preceded by an operator on the same line, which should be a violation, the `"""` is not alone on its own line. Our check already correctly flags `: """` as a violation (`Opening quotes (""") of text-block must be on the new line`), but misses `? """` .

However, the google-java-formatter reformats the code to keep both `? """` and `: """` on the same line, the formatter places the operator and `"""` together, which contradicts the style guide rule that `"""` must be on its own line.
This looks like a bug in google-java-format itself. Do you think we should report this upstream to the [google-java-format](https://github.com/google/google-java-format) ?

**Comment by mohitsatr** [comment]
> This looks like a bug in google-java-format itself

Yes, Formatting is producing the wrong output. We should totally file an issue on this.

**Comment by romani** [comment]
Please open issue on formatter.
People very unhappy on conflicts between checkstyle and formatter, and formatter usually wins, as it is automatic , enforced, and has no suppressions.

As we already do violation on such code at `:` , already conflicting. Let's be consistent and do violation for `?`.

To avoid angry user, let's create xpath suppression that will eliminate violation under `?` to stay in non conflict position. We can update coverage table that due to defect in formatter we use suppression until it fixed. 

Does is sounds like good plan?

**Comment by vivek-0509** [comment]
@romani Sounds like a good plan, will open an issue , and i think we should add xpath for both `: `and `?` since both are conflicting with google formatter

**Comment by vivek-0509** [comment]
@romani  https://github.com/google/google-java-format/issues/1378 i have created the issue

**Comment by vivek-0509** [comment]
@Zopsss @romani the issue can be moved to Google Style project..
Closing this as discussed with Mahfouz
@Atharv3221 You can reuse the work

**Comment by romani** [comment]
fix is merged
